### ISSUE TITLE
Some questions about Deformable-Convolution-V2-PyTorch for classification tasks

### ISSUE THREAD

**Issue by crocodilegogogo** [open-mmlab/mmdetection#3789]
Hi，
    I have some questions about the im2col_step parameter of the implementation. We used Deformable Convolutions in classification tasks. In the training process, we kept the training batchsize the same as im2col_step. In the test process, we put different numbers of test samples in a test batch (e.g. test the testing dataset by input one sample per time, or test the testing dataset by inputing multiple samples per time), however, we got different classification results. We also enlarged the value of im2col_step and repeated the above testing process for several times, the same experimental phenomenon can be observed. It seems that the value of im2col_step parameter and the number of samples input to the network each time impacts the final classification results. So why is this happening? What's the relationship between training and testing batchsize and im2col_step? Please kindly give me some advice. Wish for your response!

**Comment by RyanXLi** [comment]
im2col_step is used to batch process the im2col operation inside a mini-batch. It is simply for speeding up the operation. So I believe it should always be smaller than the batch size and you probably don't need to change it yourself.

**Comment by crocodilegogogo** [comment]
> 
> 
> im2col_step is used to batch process the im2col operation inside a mini-batch. It is simply for speeding up the operation. So I believe it should always be smaller than the batch size and you probably don't need to change it yourself.

Thanks very much for your kind reply,
    Since we apply Deformable in signal classification tasks (can be regarded as image classification, the y dimention is 1, we thus change the 3*3 conv2d to 1*3), we need to test multiple testing samples per Epoch. Even if we do not change the im2col parameter and keep it the default value (64) or a value smaller than the batch size, however, the classification results changes with the adjustments of testing batchsize (the accs change from 40% to 95%). We made extensive experiments and found that if the im2col_step parameter is set to 1, the testing results will not be influenced by the testing batchsize and seem correct.  Thus we are confused since the im2col_step is only a GPU speed up parameter, why does it influence the testing results? If we just set the im2col_step to 1, are the classification results correct?

                                                                                                                                                                     Wish for your kindly reply!

**Comment by crocodilegogogo** [comment]
Thanks very much for your kind reply,
Since we apply Deformable in signal classification tasks (can be regarded as image classification, the y dimention is 1, we thus change the 3×3 conv2d to 1×3), we need to test multiple testing samples per Epoch. Even if we do not change the im2col parameter and keep it the default value (64) or a value smaller than the batch size, however, the classification results changes with the adjustments of testing batchsize (the accs change from 40% to 95%). We made extensive experiments and found that if the im2col_step parameter is set to 1, the testing results will not be influenced by the testing batchsize and seem correct. Thus we are confused since the im2col_step is only a GPU speed up parameter, why does it influence the testing results? If we just set the im2col_step to 1, are the classification results correct?
Wish for your kindly reply!
